azurerm/resource_arm_virtual_network_gateway_connection.go

```diff
@@ -339,3 +478,77 @@ func resourceGroupAndVirtualNetworkGatewayConnectionFromId(virtualNetworkGateway
 
 	return resGroup, name, nil
 }
+
+func expandArmVirtualNetworkGatewayConnectionIpsecPolicies(ipsec_policies []interface{}) *[]network.IpsecPolicy {
+	ipsecPolicies := make([]network.IpsecPolicy, 0, len(ipsec_policies))
+
+	for _, d := range ipsec_policies {
+		ipsec_policy := d.(map[string]interface{})
+		ipsecPolicy := &network.IpsecPolicy{}
+
+		if dh_group, ok := ipsec_policy["dh_group"].(string); ok && dh_group != "" {
+			ipsecPolicy.DhGroup = network.DhGroup(dh_group)
+		}
+
+		if ike_encryption, ok := ipsec_policy["ike_encryption"].(string); ok && ike_encryption != "" {
+			ipsecPolicy.IkeEncryption = network.IkeEncryption(ike_encryption)
+		}
+
+		if ike_integrity, ok := ipsec_policy["ike_integrity"].(string); ok && ike_integrity != "" {
+			ipsecPolicy.IkeIntegrity = network.IkeIntegrity(ike_integrity)
+		}
+
+		if ipsec_encryption, ok := ipsec_policy["ipsec_encryption"].(string); ok && ipsec_encryption != "" {
+			ipsecPolicy.IpsecEncryption = network.IpsecEncryption(ipsec_encryption)
+		}
+
+		if ipsec_integrity, ok := ipsec_policy["ipsec_integrity"].(string); ok && ipsec_integrity != "" {
+			ipsecPolicy.IpsecIntegrity = network.IpsecIntegrity(ipsec_integrity)
+		}
+
+		if pfs_group, ok := ipsec_policy["pfs_group"].(string); ok && pfs_group != "" {
+			ipsecPolicy.PfsGroup = network.PfsGroup(pfs_group)
+		}
+
+		if v, ok := ipsec_policy["sa_datasize"].(int); ok {
+			sa_datasize := int32(v)
+			ipsecPolicy.SaDataSizeKilobytes = &sa_datasize
+		}
+
+		if v, ok := ipsec_policy["sa_lifetime"].(int); ok {
+			sa_lifetime := int32(v)
+			ipsecPolicy.SaLifeTimeSeconds = &sa_lifetime
+		}
+
+		ipsecPolicies = append(ipsecPolicies, *ipsecPolicy)
+	}
+
+	return &ipsecPolicies
+}
+
+func flattenArmVirtualNetworkGatewayConnectionIpsecPolicies(ipsecPolicies *[]network.IpsecPolicy) []interface{} {
+	ipsec_policies := make([]interface{}, 0, len(*ipsecPolicies))
+
+	for _, ipsecPolicy := range *ipsecPolicies {
```

Comment:
there's a potential crash here if `ipsecPolicies` is nil - could we make this:

```
ipsec_policies := make([]interface{}, 0)
if inputPolicies := ipsecPolicies; inputPolicies != nil {
  for _, ipsecPolicy := range *inputPolicies {
    # ..
  }
}
```

Reply:
Yes, I shouldn't rely on the caller here